File: src/065. Convergents of e/065.rb
# n starts from 2
def fraction_seq(n)
  if n % 3 == 0
    n / 3 * 2
  else
    1
  end
end

e = 0
100.downto(2) do |n|
  e = Rational('1') / (fraction_seq(n) + e)
end
e += 2
puts e.numerator.to_s.chars.map(&:to_i).inject(:+)

File: src/065. Convergents of e/065.py
from fractions import Fraction

# n starts from 2
def fraction_seq(n):
    if n % 3 == 0:
        return n // 3 * 2
    else:
        return 1

e = Fraction(0)
for n in range(100, 1, -1):
    e = Fraction(1) / (fraction_seq(n) + e)

e += 2
print(sum(map(int, list(str(e.numerator)))))
